spectrum/trd/disk_info.go

```diff
@@ -0,0 +1,62 @@
+package trd
+
+import (
+	"fmt"
+	"github.com/mrcook/retroio/storage"
+)
+
+// DiskInformation stores the disk information obtained from the 9th sector of the system track
+type DiskInformation struct {
```

Comment:
Would you consider updating the struct and reader to also include the unused atributes? This would help both for documentation, and especially for when I eventually get round to implemeting the writers for the media formats. For example:

```
type DiskInformation struct {
	EndOfCatalogueInfo uint8      // :=0 - Signals end of catalogue (?)
	Unused1            [224]uint8 // Unused (filled with zero - 0x00)
	NextFreeSector     uint8      // First free sector address: sec
	NextFreeTrack      uint8      // First free sector address: track
	DiskType           DiskType   // Disk type (0x16..0x19)
	NumFiles           uint8      // Number of files on disk
	NumFreeSectors     uint16     // Number of free sectors on disk
	TrdId              uint8      // :=0x10 - TR-DOS ID
	Unused2            uint8      // Unused: 0x00
	Unused3            [9]uint8   // Unused: 0x20 (space)
	Unused4            uint8      // Unused: 0x00
	NumDeletedFiles    uint8      // Number of deleted files on disk
	Label              [8]byte    // Disk label
	EndOfDiskInfo      [3]uint8   // End of disk info, filled with 0x00 (Unused?)
}
```

Reply:
While I agree that it's better from the documentation standpoint, using this approach will block the further adoption of the [SCL](https://faqwiki.zxnet.co.uk/wiki/SCL_format) format (if you're interested). It effectively describes the same resulting TRD image but in a more compact format that omits irrelevant some details.

Additionally, if/when it comes to unit testing of the presentation-level code, having to mock these reader-related details would also create unnecessary burden.

Reply:
On the other hand, what you documented above, is the structure of the system sector (not disk info). It indeed can be documented and implemented like this and then produce a "pure" disk information struct.

Reply:
I only briefly looked at the documentation you linked, so I thought the `DiskInformation` you used was the same. 

Can not the SCL images be considered a different disk format? As the `file begins with a 8 byte 'magic': SINCLAIR` I guess it can be handled so no?

If you feel it's better so leave it as-is please do so.

Reply:
> Can not the SCL images be considered a different disk format?

I believe it's fair to say that the difference between TRD and SCL is similar to the difference between TZX and TAP respectively. The former describes more physical details of the medium but in most they are irrelevant. Therefore, while both TRD and SCL are different formats, they represent the same underlying model (TR-DOS disk).

Actually, if sourced from an SCL image, the disk information would look like the following:
```go
type DiskInformation struct {
	NumFiles uint8
}
```

> If you feel it's better so leave it as-is please do so.

I believe that `Unused1..4` represent the `DiskInformation` struct, and it's better to leave it as is for now. Often, the need for refactoring becomes more obvious when new requirements get introduced (e.g. convert tape to disk), and at that time there's more information to make the right choice.